test/applications_unit_test.go

```diff
@@ -165,9 +168,52 @@ func getSuggestedParams(
 	}, err
 }
 
+func weMakeAGetAssetByIDCall(assetID int) error {
+	clt, err := algod.MakeClient(mockServer.URL, "")
+	if err != nil {
+		return err
+	}
+	clt.GetAssetByID(uint64(assetID)).Do(context.Background())
```

Comment:
Looks like these tests save the error for additional validation:
`_, globalErrForExamination = algodClient.Block(uint64(blocknum)).Do(context.Background())`

Reply:
globalErrForExamination plays a role in scenario:
Then expect error string to contain "<err>"
The feature file for responses does not have that scenario.

Reply:
Thats with the old response tests format, the new ones should actually check the error response against these:
```
resources/generated_responses/algod_applications_ErrorResponse_0.json
resources/generated_responses/indexer_applications_ErrorResponse_0.json
```